docs/apriori_sbms.ipynb

```diff
@@ -0,0 +1,384 @@
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import sys\n",
+    "import matplotlib.pyplot as plt\n",
+    "sys.path.append(\"../\")\n",
+    "import pandas as pd\n",
+    "from graph import GraphIO\n",
+    "from analysis.apriori_sbm_code import fit_apriori_sbm"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "## Ciona Intestinalis ##\n",
+    "We'll start with the single Ciona Connectome. First we load the data:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "path = '../json_connectomes/ciona.json'\n",
+    "ciona_connectome, _, _, _ = GraphIO.load(path)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "Now we'll fit an SBM for each grouping"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# In Ciona, Hemisphere is called side and cell_type is called annotation\n",
+    "groups = [['Side'], ['Annotation'], ['Side', 'Annotation']]\n",
+    "rows = {'Ciona Intestinalis': {}}\n",
+    "for group in groups:\n",
+    "    rows['Ciona Intestinalis'][str(group)] = fit_apriori_sbm(ciona_connectome, group, plot_sbms=False)\n",
+    "ciona_results = pd.DataFrame.from_dict({(i,j): rows[i][j]\n",
+    "                           for i in rows.keys()\n",
+    "                           for j in rows[i].keys()},\n",
+    "                       orient='index')\n",
+    "ciona_results.head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "Great, now lets plot the BIC for different groupings:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "fig, ax = plt.subplots(2, 1)\n",
+    "out_index = ciona_results.index.levels[0]\n",
+    "for i, key in enumerate(out_index):\n",
+    "    data = ciona_results.loc[key]\n",
+    "    ax[0].scatter(data['bic'].index, data['bic'], label=key)\n",
+    "    ax[1].scatter(data['liklihood'].index, data['liklihood'], label=key)\n",
+    "    ax[0].set(xlabel='Group', ylabel='-BIC')\n",
+    "    ax[1].set(xlabel='Group', ylabel='Likelihood')\n",
+    "fig.suptitle('BIC and Likelihood: Witvilet at Different Developmental Stages')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "It appears annotation has the best BIC but the worst liklihood. This could be because annotation has much fewer parameters than the other groupings. (For Ciona\n",
```

Comment:
Shouldn't this be - log(likelihood)?  Also "liklihood"? Is that an intentional abbreviation or just a typo?

Reply:
typo